sample/kotlin/src/main/kotlin/com/datadog/android/sample/data/DataRepository.kt

```diff
@@ -19,15 +20,27 @@ class DataRepository(
     private val localDataSource: LocalDataSource
 ) {
 
+    @Suppress("SimpleRedundantLet")
     fun getLogs(query: String): Flowable<List<Log>> {
         return Single.concat(
             localDataSource.fetchLogs(),
             remoteDataSource.getLogs(query)
-                .observeOn(Schedulers.io())
+                .subscribeOn(Schedulers.io())
                 .map { it.data }
                 .doOnSuccess {
                     localDataSource.persistLogs(it)
                 }
+                .doOnSubscribe {
+                    val span = GlobalTracer.get()
```

Comment:
I'm wondering if we could do this automatically somehow, because it is essentially `subscribe-dispose` lifecycle.

Say I have a chain with multiple requests inside, so for every request I will need to do that?

UPD: ok, after reading the next commit I think I got it: we just need to do this on the top-most level on the stream once, and not on the child streams?

Reply:
Exactly and in any case those are extra helpers that they can write them selfs if they need more encapsulation. We jus need to provide the base tools.